### PR TITLE
drivers: apds9960: Use dts generated I2C name for binding

### LINKED ISSUE #7804: samples/sensor/apds9960 fails with assertion "Fatal fault in essential thread" on quark_se_c1000_ss_devboard
The test **samples/sensor/apds9960** results into exception on arc core.

**Arch:** arc
**Board:** quark_se_c1000_ss_devboard
**Zephyr tag:**  v1.12.0-rc1 - commit e7509c17c5938b48f9ad4b7ae78199bb0faa6113.
**Pre-requisite:** Sensor connected on I2C SCL, SDA pins of arc core.

**Error:**
```
****** Booting Zephyr OS v1.12.0-rc1-41-gfcc56e3 *****
APDS9960 sample application
Exception vector: 0x3, cause code: 0x4, parameter 0x0
Address 0x40002248
Current thread ID = 0xa8000508
Faulting instruction address = 0x40002248
Fatal fault in essential thread! Spinning...
```

The test was working fine till zephyr-v1.11.0 tag.

**Steps to reproduce:**
```
rm -rf build && mkdir build && cd build
cmake -D BOARD=quark_se_c1000_ss_devboard ../
make BOARD=quark_se_c1000_ss_devboard flash

Check the console log.
```

**Expected output:**
```
***** BOOTING ZEPHYR OS v1.11.0 - BUILD: May 23 2018 20:29:24 *****                                                   
APDS9960 sample application                                                                                           
ambient light intensity without trigger is 0
proxy without trigger is 42
ambient light intensity without trigger is 211
```

### PR DESCRIPTION
**Pull request by punitvara**
This commit removes old Kconfig dependency on I2C name and
replaced with new dts generated I2C name.

Fixes #7804

This issue was produced because of
9ddd8d1e4d7daa9b1d37af0e8a23de1c1900d7d5

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Review by SavinayDharmappa — APPROVED** [review]
LGTM

File: drivers/sensor/apds9960/Kconfig [hunk 1]
```diff
@@ -14,7 +14,6 @@ menuconfig APDS9960
 config  APDS9960_I2C_DEV_NAME
 	string "I2C Master"
 	depends on APDS9960
-	default I2C_SS_0_NAME
```
Comment (SavinayDharmappa) [inline drivers/sensor/apds9960/Kconfig]: remove this Kconfig . It is not neccessary. Otherwise ok.

**Review by galak — CHANGES_REQUESTED** [review]
This change ends up making the driver us a name/define specific to a given SoC.  That isn't the direction we should be going.  Instead if you want to use this from .dts we do a fixup in the board dts.fixup.

Look at FXOS8700 for an example

File: drivers/sensor/apds9960/apds9960.c [hunk 2]
```diff
@@ -458,11 +458,11 @@ static int apds9960_init(struct device *dev)
 	struct apds9960_data *data = dev->driver_data;
 	int als_gain = 0;
 
-	data->i2c = device_get_binding(CONFIG_APDS9960_I2C_DEV_NAME);
+	data->i2c = device_get_binding(CONFIG_I2C_SS_0_NAME);
```
Comment (nashif) [inline drivers/sensor/apds9960/apds9960.c]: this does not look right, this is a generic sensor that should work with any board or configuration, yet you make it depend on I2C_SS, i.e. arduino_101 and quark se. What we had before is a actually correct.

**Review by nashif — DISMISSED** [review]

File: drivers/sensor/apds9960/apds9960.c (hunk 2 above)
Comment (punitvara) [reply drivers/sensor/apds9960/apds9960.c]: ohh yeah I got you

**Review by galak — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**